### PR TITLE
new parameters in `plot_bloch_multivector`: `figsize`, `font_size`, and `title_font_size` 

### LINKED ISSUE #7263: Better use for the `figsize` argument of `plot_bloch_multivector`
### What is the expected enhancement?

The `figsize` argument of `plot_bloch_multivector` is currently not used by the visualisation, so it is impossible to change its size (e.g. to shrink it for single-qubit states). The following tiny change to its code would use `figsize` as a hint for individual bloch ball sizes, enhancing the usability of this visualization:

```py
# Original source code snippet:
...
width, height = plt.figaspect(1 / num)
...

# Modified source code snippet:
...
if figsize is not None:
    width, height = figsize
    width *= num
else:
    width, height = plt.figaspect(1 / num)
...
```

Comparison of a single-qubit bloch sphere visualisation, original (left) vs scaled version available in proposed enhancement (right):

image

Three-qubit bloch spheres, original (top) vs scaled version available in proposed enhancement (bottom):

image

Three-qubit bloch spheres, advanced scaled version available in proposed enhancement, leaving additional space between spheres:

image

Full source code for reference:

```py
def plot_bloch_multivector(state, title="", figsize=None, *, rho=None, reverse_bits=False):
    if not HAS_MATPLOTLIB:
        raise MissingOptionalLibraryError(
            libname="Matplotlib",
            name="plot_bloch_multivector",
            pip_install="pip install matplotlib",
        )
    from matplotlib import pyplot as plt
    bloch_data = (
        _bloch_multivector_data(state)[::-1] if reverse_bits else _bloch_multivector_data(state)
    )
    num = len(bloch_data)
    if figsize is not None:
        width, height = figsize
        width *= num
    else:
        width, height = plt.figaspect(1 / num)
    fig = plt.figure(figsize=(width, height))
    for i in range(num):
        pos = num - 1 - i if reverse_bits else i
        ax = fig.add_subplot(1, num, i + 1, projection="3d")
        plot_bloch_vector(bloch_data[i], "qubit " + str(pos), ax=ax, figsize=figsize)
    fig.suptitle(title, fontsize=16)
    matplotlib_close_if_inline(fig)
    return fig
```

### PR DESCRIPTION
**Pull request by sg495**
### Summary

Closes #7263, introducing a use for the `figsize` argument of `plot_bloch_multivector`.

### Details and comments

Previously, the `figsize` argument of `plot_bloch_multivector` was not used by the visualisation, so it was impossible to change its size (e.g. to shrink it for single-qubit states). With this pull request, `figsize` can be used to set a size for individual Bloch ball sub-plots: if there are `num` qubits and `(w, h)` is passed to `figsize`, then the overall figure width is set to `num*w`, while its overall height is set to `h`.

### CODE REVIEW

**Comment by sg495** [comment]
I have made a few updates to the original pull request on `plot_bloch_multivector`.

Introduced a `font_size` kwarg to control font size in Bloch sphere plots:

image

If custom `font_size` is used, title font size is adjusted accordingly:

image

Introduced a `title_font_size` and `title_pad` kwargs to control title font size and adjust padding:

image

File: qiskit/visualization/state_visualization.py [hunk 1]
```diff
@@ -198,6 +200,7 @@ def plot_bloch_vector(bloch, title="", ax=None, figsize=None, coord_type="cartes
         figsize (tuple): Figure size in inches. Has no effect is passing ``ax``.
         coord_type (str): a string that specifies coordinate type for bloch
             (Cartesian or spherical), default is Cartesian
+        font_size (double): Font size.
```
Comment (1ucian0) [inline qiskit/visualization/state_visualization.py:203]:
```suggestion
        font_size (float): Font size.
```

File: qiskit/visualization/state_visualization.py [hunk 2]
```diff
@@ -250,8 +262,11 @@ def plot_bloch_multivector(
     Args:
         state (Statevector or DensityMatrix or ndarray): an N-qubit quantum state.
         title (str): a string that represents the plot title
-        figsize (tuple): Has no effect, here for compatibility only.
+        figsize (tuple): size of each individual Bloch sphere figure, in inches.
         reverse_bits (bool): If True, plots qubits following Qiskit's convention [Default:False].
+        font_size (double): Font size for the Bloch ball figures.
+        title_font_size (double): Font size for the title.
+        title_pad (double): Padding for the title (suptitle `y` position is `y=1+title_pad/100`).
```
Comment (1ucian0) [inline qiskit/visualization/state_visualization.py:269]:
```suggestion
        font_size (float): Font size for the Bloch ball figures.
        title_font_size (float): Font size for the title.
        title_pad (float): Padding for the title (suptitle `y` position is `y=1+title_pad/100`).
```

**Review by 1ucian0 — CHANGES_REQUESTED** [review]
this looks good! Considering adding some testing in `test/ipynb/mpl/graph/test_graph_matplotlib_drawer.py`

File: qiskit/visualization/bloch.py [hunk 3]
```diff
@@ -130,7 +130,9 @@ class Bloch:
             Positions of +z and -z labels respectively.
     """
 
-    def __init__(self, fig=None, axes=None, view=None, figsize=None, background=False):
+    def __init__(
+        self, fig=None, axes=None, view=None, figsize=None, background=False, font_size=None
```
Comment (1ucian0) [inline qiskit/visualization/bloch.py:134]:
```suggestion
        self, fig=None, axes=None, view=None, figsize=None, background=False, font_size=20
```

**Comment by sg495** [comment]
> this looks good! Considering adding some testing in `test/ipynb/mpl/graph/test_graph_matplotlib_drawer.py`
> 
> and a release note! Check the contributor guidelines https://github.com/Qiskit/qiskit-terra/blob/main/CONTRIBUTING.md

I have included a release note and added a test for the new features, as requested 👍

File: test/ipynb/mpl/graph/test_graph_matplotlib_drawer.py [hunk 4]
```diff
@@ -370,6 +370,31 @@ def test_plot_coupling_map(self):
...
+
+        self.graph_state_drawer(
+            state=state,
+            output="bloch",
+            figsize=(3,2),
+            font_size=10,
+            title="|0+R> state",
+            title_font_size=14,
+            title_pad=8,
+            filename="bloch_multivector_figsize_improvements.png"
```
Comment (1ucian0) [inline test/ipynb/mpl/graph/test_graph_matplotlib_drawer.py:395]: This reference file needs to exist in `test/ipynb/mpl/graph/references`.

**Review by 1ucian0 — CHANGES_REQUESTED** [review]
The test needs a reference file. Ill try to add it myself.

File: test/ipynb/mpl/graph/test_graph_matplotlib_drawer.py [hunk 5]
```diff
@@ -370,6 +370,31 @@ def test_plot_coupling_map(self):
...
         )
 
+    def test_plot_bloch_multivector_figsize_improvements(self):
+        """test bloch sphere figsize, font_size, title_font_size and title_pad
+        See https://github.com/Qiskit/qiskit-terra/issues/7263
+        and https://github.com/Qiskit/qiskit-terra/pull/7264.
+        """
+        circuit = QuantumCircuit(3)
+        circ.h(1)
+        circ.sxdg(2)
```
Comment (1ucian0) [inline test/ipynb/mpl/graph/test_graph_matplotlib_drawer.py:380]: This is `NameError`, I think
```suggestion
        circuit.h(1)
        circuit.sxdg(2)
```

**Review by 1ucian0 — APPROVED** [review]
Thanks for the code and the patience! I fixed some things in the tests ([have a look](https://github.com/Qiskit/qiskit-terra/pull/7264/commits/356e98e57b2a819c94e61a20764f655bc643c87e) if curious). Next time, consider making smaller PRs, once per feature/bugfix, so they can be reviewed independently.

**Review by 1ucian0 — APPROVED** [review]
Thanks for the code and the patience! I fixed some things in the tests ([have a look](https://github.com/Qiskit/qiskit-terra/pull/7264/commits/356e98e57b2a819c94e61a20764f655bc643c87e) if curious). Next time, consider making smaller PRs, once per feature/bugfix, so they can be reviewed independently (and the changelog would get created more nicely)

**Pull request merged**